record.c

```diff
@@ -248,8 +248,16 @@ int janus_recorder_close(janus_recorder *recorder) {
 		/* We need to rename the file, to remove the temporary extension */
 		char newname[1024];
 		memset(newname, 0, 1024);
-		g_snprintf(newname, strlen(recorder->filename)-strlen(rec_tempext), "%s", recorder->filename);
-		if(rename(recorder->filename, newname) != 0) {
+		char tmpname[1024];
+		memset(tmpname, 0, 1024);
+		if(recorder->dir) {
+			g_snprintf(newname, strlen(recorder->filename)-strlen(rec_tempext), "%s/%s", recorder->dir, recorder->filename);
```

Comment:
Yes, just realised I need to recalculate the length based on the recording dir also?